### PR TITLE
kata 1 to four

### PR DESCRIPTION
**Pull request by benjspeight**

### CODE REVIEW

File: test/kata1.fizzBuzz.test.js [hunk 1]
```diff
@@ -1,11 +1,18 @@
 const { fizzBuzz } = require("../src");
 
 describe("fizzBuzz", () => {
-  test("returns Fizz when passed a multiple of 3", () => {});
+  it("returns Fizz when passed a multiple of 3", () => {});
+    expect(fizzBuzz(3)).toBe('Fizz')
```
Comment (szabinakov) [inline test/kata1.fizzBuzz.test.js:5]: The test cases' expect blocks are not in the correct position in this file. 
They are after the it block instead of being in the curly braces. 
The tests will still pass if they are correct but they wont  belong to the correct block, with the description.
Should be like this:

`````
it("returns Fizz when passed a multiple of 3", () => {
 expect(fizzBuzz(3)).toBe('Fizz')
});
`````

File: test/kata1.fizzBuzz.test.js [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
 
-  test("returns Buzz when passed a multiple of 5", () => {});
+  it("returns Buzz when passed a multiple of 5", () => {});
+  expect(fizzBuzz(5)).toBe('Buzz')
```
Comment (szabinakov) [inline test/kata1.fizzBuzz.test.js:8]: Same as above, expect is in the wrong place

File: test/kata1.fizzBuzz.test.js [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
 
-  test("returns FizzBuzz when passed a multiple 3 and 5", () => {});
+  it("returns FizzBuzz when passed a multiple 3 and 5", () => {});
+  expect(fizzBuzz(15)).toBe('FizzBuzz')
```
Comment (szabinakov) [inline test/kata1.fizzBuzz.test.js:11]: Same as above all these expects are in the wrong place

File: test/kata1.fizzBuzz.test.js [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
+  expect(fizzBuzz(30)).toBe('FizzBuzz')
+  expect(fizzBuzz(90)).toBe('FizzBuzz')
 
-  test("returns the number when it isn't a multiple of 3 or 5", () => {});
+  it("returns the number when it isn't a multiple of 3 or 5", () => {});
+  expect(fizzBuzz(4)).toBe(4)
```
Comment (szabinakov) [inline test/kata1.fizzBuzz.test.js:16]: Expects are in the wrong place

File: test/kata2.booleanToWord.test.js [hunk 5]
```diff
@@ -1,5 +1,12 @@
...
 describe("booleanToWord", () => {
-  // how do we create specs again???
+  test("returns Yes when passed true", () => {
+    expect(booleanToWord(true)).toBe("Yes");
+  });
+
+  test("returns No when passed false", () => {
+    expect(booleanToWord(false)).toBe("No");
+  });
+
```
Comment (szabinakov) [inline test/kata2.booleanToWord.test.js:11]: This extra line can be deleted for better formatting.

File: src/kata2.booleanToWord.js [hunk 6]
```diff
@@ -1,3 +1,9 @@
-const booleanToWord = (boolean) => {};
+const booleanToWord = (boolean) => {
+    if (boolean) {
```
Comment (szabinakov) [inline src/kata2.booleanToWord.js:2]: It can be refactored by using the ternary operator. 
`````
const booleanToWord = (boolean) => (boolean ? "Yes" : "No")
`````
The conditional (ternary) operator takes three operands: a condition followed by a question mark (?), then an expression to execute if the condition is truthy followed by a colon (:), and finally the expression to execute if the condition is falsy. This operator is frequently used as an alternative to an if...else  statement.

File: src/kata3.numberToReversedDigits.js [hunk 7]
```diff
@@ -1,3 +1,6 @@
-const numberToReversedDigits = (number) => {};
+const numberToReversedDigits = (number) => {
+const numArray = number.toString().split("").reverse().map(Number);
+  return numArray;
```
Comment (szabinakov) [inline src/kata3.numberToReversedDigits.js:3]: Instead of assigning the method to a variable you could just return it in one line.
`````
return number.toString().split("").reverse().map(Number);
``````

File: test/kata1.fizzBuzz.test.js [hunk 8]
```diff
@@ -1,11 +1,18 @@
 const { fizzBuzz } = require("../src");
 
 describe("fizzBuzz", () => {
-  test("returns Fizz when passed a multiple of 3", () => {});
+  it("returns Fizz when passed a multiple of 3", () => {});
```
Comment (szabinakov) [inline test/kata1.fizzBuzz.test.js:4]: Maybe it would also be nice to add some tests for negative numbers etc... Just in general to think about edge cases, trying to break your code, to make it more secure.

At some places you use "" and some places you use ''. For better consistency I would stick to "".

File: test/kata4.humanCatDogYears.test.js [hunk 9]
```diff
@@ -1,3 +1,21 @@
...
+    expect(humanCatDogYears(3)).toEqual([3, 28, 29]);
+    expect(humanCatDogYears(4)).toEqual([4, 32, 34]);
+    expect(humanCatDogYears(5)).toEqual([5, 36, 39]);
+    expect(humanCatDogYears(6)).toEqual([6, 40, 44]);
+    expect(humanCatDogYears(7)).toEqual([7, 44, 49]);
+    expect(humanCatDogYears(8)).toEqual([8, 48, 54]);
+    expect(humanCatDogYears(9)).toEqual([9, 52, 59]);
+    expect(humanCatDogYears(10)).toEqual([10, 56, 64]);
+  });
+
```
Comment (szabinakov) [inline test/kata4.humanCatDogYears.test.js:16]: Lots of extra lines could be deleted for better formatting.

File: src/kata4.humanCatDogYears.js [hunk 10]
```diff
@@ -1,3 +1,22 @@
-const humanCatDogYears = (number) => {};
+const humanCatDogYears = (number) => {
+    let catYears = 0;
+    let dogYears = 0;
+    let humanYears = number;
```
Comment (szabinakov) [inline src/kata4.humanCatDogYears.js:4]: The human years is a constant variable and never being reassigned, therefor it should be const instead of let.

File: test/kata6.joinNames.test.js [hunk 11]
```diff
@@ -1,5 +1,17 @@
 const { joinNames } = require("../src");
 
 describe("joinNames", () => {
-  test("returns string of names, seperated by commas and an ampersand", () => {});
+  test("returns string of names, seperated by commas and an ampersand", () => {
```
Comment (szabinakov) [inline test/kata6.joinNames.test.js:4]: There is a typo, separated* :)

File: test/kata6.joinNames.test.js [hunk 12]
```diff
@@ -1,5 +1,17 @@
...
+  const names1 = [{name: 'Homer'}, {name: 'Marge'}, {name: 'Bart'}, {name: 'Lisa'}, {name: 'Maggie'}];
```
Comment (szabinakov) [inline test/kata6.joinNames.test.js:5]: When there is an array of objects and it is a bit long, you could reformat it like this:
`````
    const names1 = [
      { name: "Homer" },
      { name: "Marge" },
      { name: "Bart" },
      { name: "Lisa" },
      { name: "Maggie" },
    ];
`````
To make it look a little bit tidier, and easier to read.
Also at some place you use '' and at some place you are using "". For better consistency I would recommend to use "" everywhere.

File: src/kata3.numberToReversedDigits.js [hunk 13]
```diff
@@ -1,3 +1,6 @@
-const numberToReversedDigits = (number) => {};
+const numberToReversedDigits = (number) => {
+const numArray = number.toString().split("").reverse().map(Number);
```
Comment (szabinakov) [inline src/kata3.numberToReversedDigits.js:2]: missing indentation.

File: src/kata5.reachDestination.js [hunk 14]
```diff
@@ -1,3 +1,8 @@
-const reachDestination = (distance, speed) => {};
+const reachDestination = (distance, speed) => {
+const reachTime = distance / speed;
```
Comment (szabinakov) [inline src/kata5.reachDestination.js:2]: Line 2,3,4,5 are missing indentation

File: test/kata7.getEmployerRole.test.js [hunk 15]
```diff
@@ -1,5 +1,21 @@
 const { getEmployerRole } = require("../src");
 
+const employees = [
+  {
+    name: 'Satti',
```
Comment (szabinakov) [inline test/kata7.getEmployerRole.test.js:5]: I would recommend to use "" - double quotations. You used doubles throughout the application, also it is recommended to use "" and that is because as you can see on line 19, single quote is also used in strings. (employee's)

File: src/kata7.getEmployerRole.js [hunk 16]
```diff
@@ -1,3 +1,12 @@
-const getEmployerRole = (employeeName, employees) => {};
+const getEmployerRole = (employeeName, employees) => {
+    
+      for (let i = 0; i < employees.length; i++) {
+        if (employees[i].name === employeeName) {
+          return employees[i].role;
+        }
+      }
+      return "Employee not found";
+
```
Comment (szabinakov) [inline src/kata7.getEmployerRole.js:9]: The extra line could be deleted (line 1 and line 9). 
The for loop's indentation is a bit too much, could be less like this:

`````
const getEmployerRole = (employeeName, employees) => {
  for (let i = 0; i < employees.length; i++) {
    if (employees[i].name === employeeName) {
      return employees[i].role;
    }
  }
  return "Employee not found";
};

module.exports = getEmployerRole;
``````

**Review by szabinakov — COMMENTED** [review]
All the tests are passing which is the overall goal :) well done, nice work :)
I have added some comments below it is mainly on the formatting. The indentation is off on a lots of pages. There are extra lines left in the code as well and inconsistency with the '' and "" quotation marks. I would suggest to stick with the "" and that is because the single quotation marks are also used in strings for example: "Tom's mom" would work but 'Tom's mom' would not work because the middle ' would close the string.
